app/Observers/Immobilisation/AssetMaintenanceTicketObserver.php

```
@@ -0,0 +1,31 @@
+use App\Models\Immobilisation\AssetMaintenanceTicket;
+use Illuminate\Support\Facades\DB;
+
+class AssetMaintenanceTicketObserver
+{
+    public function creating(AssetMaintenanceTicket $ticket): void
+    {
+        if (empty($ticket->reference)) {
+            $ticket->reference = DB::transaction(function () {
+                $year = now()->year;
+
+                $latest = AssetMaintenanceTicket::whereYear('created_at', $year)
```

PHPStan (2.2.7)

[error] Call to an undefined static method App\Models\Immobilisation\AssetMaintenanceTicket::whereYear().


(staticMethod.notFound)